blockchain/api.go

```diff
@@ -1322,6 +1446,31 @@ func (api *OracleUSDAPI) SetCurrentPrice(ctx context.Context, key *ecdsa.Private
 	return api.oracleContract.SetCurrentPrice(opts, price)
 }
 
+func (api *OracleUSDAPI) PackSetCurrentPriceTransactionData(price *big.Int) ([]byte, error) {
+	oracleABI, err := abi.JSON(bytes.NewBufferString(marketAPI.OracleUSDABI))
+	if err != nil {
+		return nil, err
+	}
+	return oracleABI.Pack("SetCurrentPrice", price)
+}
+
+func (api *OracleUSDAPI) UnpackSetCurrentPriceTransactionData(data []byte) (*big.Int, error) {
+	if len(data) == 0 {
+		return nil, fmt.Errorf("data is empty")
+	}
+
+	if len(data) < 36 {
+		return nil, fmt.Errorf("data is malformed")
+	}
+	// Cut method specify byte, stands only parameters data
+	// Method ids are saves at the first 4 bytes of the hash of the
+	// methods string signature. (signature = baz(uint32,string32))
+	params := data[4:]
+
+	price := common.HexToAddress(common.Bytes2Hex(params[:32])).Big()
```

Comment:
why not using just `data[4:32]`?

Reply:
its just for separate method and parameters.
more visual effect, for developers. =)